src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImplTest.java

```diff
@@ -29,76 +42,116 @@ class CompanyPscsServiceImplTest {
     private static final String COMPANY_NUMBER = "12345678";
     private static final String ENCODED_VALUE = "ENCODED";
     private static final String ETAG = "ETAG";
+    private static final String INDIVIDUAL = "individual-person-with-significant-control";
+    private static final Instant dateNowInstant = Instant.parse("2023-06-01T00:00:00Z");
 
     @Mock
     private CompanyPscsRepository repository;
     @Mock
     private RandomService randomService;
+    @Mock
+    private Clock clock;
 
     @InjectMocks
     private CompanyPscsServiceImpl companyPscsService;
 
     @Test
     void create() throws DataException {
+
         CompanySpec spec = new CompanySpec();
         spec.setCompanyNumber(COMPANY_NUMBER);
 
-        when(this.randomService.getString(30)).thenReturn(ENCODED_VALUE);
-        when(this.randomService.getEncodedIdWithSalt(ID_LENGTH, SALT_LENGTH)).thenReturn(ENCODED_VALUE);
-
-        when(this.randomService.getEtag()).thenReturn(ETAG);
         CompanyPscs savedPsc = new CompanyPscs();
         when(this.repository.save(any())).thenReturn(savedPsc);
 
+        when(this.clock.instant()).thenReturn(dateNowInstant);
+        when(this.randomService.getEncodedIdWithSalt(ID_LENGTH, SALT_LENGTH)).thenReturn(ENCODED_VALUE);
+        when(this.randomService.getEtag()).thenReturn(ETAG);
+        when(randomService.getNumberInRange(0, NATURES_OF_CONTROL.length))
+                .thenReturn(OptionalLong.of(2L), OptionalLong.of(1L), OptionalLong.of(4L));
+
         CompanyPscs returnedPsc = this.companyPscsService.create(spec);
 
         assertEquals(savedPsc, returnedPsc);
 
-        ArgumentCaptor<CompanyPscs> pscCaptor = ArgumentCaptor.forClass(CompanyPscs.class);
-        verify(repository).save(pscCaptor.capture());
+        verify(repository).save(argThat(this::hasExpectedValues));
+    }
+
+    private boolean hasExpectedValues(CompanyPscs companyPsc) {
```

Comment:
Possible refactor this long method with methods like `hasExpectedLocaton`, `hasExpectedNaturesOfControl` etc.